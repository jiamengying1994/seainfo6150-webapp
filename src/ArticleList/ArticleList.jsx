import React from "react";
import PropTypes from "prop-types";
import ArticleListItem from "../ArticleListItem/ArticleListItem";
import styles from "./ArticleList.module.css";

const ArticleList = (props) => {
  const articles = props.articleList;

  return (
    <div className={styles.outer}>
     <div className = {styles.grid} >
      {articles.map((articleListItem) => (

      <div className={styles.box} key  = {articleListItem.slug}>
        <ArticleListItem
        articleListItem = {articleListItem}/>
      </div>
      ))
      }
    </div>
    </div>

  );
};

export default ArticleList;