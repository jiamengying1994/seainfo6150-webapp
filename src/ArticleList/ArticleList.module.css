.box {
display: flex;
flex-direction: column;
justify-content: space-between;
border: red;
border-style: solid;
border-width: 1px;
background: rgb(245,114,121);
background: linear-gradient(90deg, rgba(245,114,121,1) 0%, rgba(255,255,255,1) 17%);
margin: 20px;
border-radius: 10px;
padding-left:5px;
padding-right:5px;
padding-bottom:5px;
}

.outer{
box-shadow: 0px 0px 6px 5px rgba(255,105,105,0.63);
padding:10px;
background:grey;
border-radius: 10px;
}

.grid{
display: grid;
grid-template-columns: 1fr 1fr 1fr;
}