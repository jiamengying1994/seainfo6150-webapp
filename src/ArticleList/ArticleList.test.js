import React from "react";
import ArticleList from "./ArticleList";
import { fireEvent } from "@testing-library/react";

describe("ArticleList tests", () => {


  it("renders correctly after button is clicked", () => {
    const al = [{"title": "1"}, {"title": "2"},]
    const { container } = render(
      <ArticleList articleList={al}/>
    );

    expect(container).toMatchSnapshot();
  });
});