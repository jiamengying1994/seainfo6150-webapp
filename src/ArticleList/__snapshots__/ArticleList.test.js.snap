// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders correctly after button is clicked 1`] = `
<div>
  <div
    class="outer"
  >
    <div
      class="grid"
    >
      <div
        class="box"
      >
        <div>
          <h2
            class="title"
          >
            1
          </h2>
          <p
            class="date"
          />
          
          <p />
        </div>
        <button
          class="button"
        >
          Show more
        </button>
      </div>
      <div
        class="box"
      >
        <div>
          <h2
            class="title"
          >
            2
          </h2>
          <p
            class="date"
          />
          
          <p />
        </div>
        <button
          class="button"
        >
          Show more
        </button>
      </div>
    </div>
  </div>
</div>
`;
